cmd/abapEnvironmentCreateSystem.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/SAP/jenkins-library/pkg/abaputils"
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/ghodss/yaml"
+)
+
+func abapEnvironmentCreateSystem(config abapEnvironmentCreateSystemOptions, telemetryData *telemetry.CustomData) {
+
+	cf := cloudfoundry.CFUtils{Exec: &command.Command{}}
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runAbapEnvironmentCreateSystem(&config, telemetryData, cf)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapEnvironmentCreateSystem(config *abapEnvironmentCreateSystemOptions, telemetryData *telemetry.CustomData, cf cloudfoundry.CFUtils) error {
+
+	if config.ServiceManifest != "" {
+		// if the manifest file is provided, it is directly passed through to cloudFoundryCreateService
+		createServiceConfig := cloudFoundryCreateServiceOptions{
+			CfAPIEndpoint:   config.CfAPIEndpoint,
+			CfOrg:           config.CfOrg,
+			CfSpace:         config.CfSpace,
+			Username:        config.Username,
+			Password:        config.Password,
+			ServiceManifest: config.ServiceManifest,
+		}
+		runCloudFoundryCreateService(&createServiceConfig, telemetryData, cf)
```

Comment:
@CCFenner @OliverNocon I'm reusing this method from cloudFoundryCreateService. Is this ok or should I avoid this? Or move the whole implementation to utils package?